etl/tg-jdbc-driver/tg-jdbc-restpp/src/main/java/com/tigergraph/jdbc/RestppResultSet.java

```diff
@@ -139,10 +147,13 @@ private void parseKeyValue(ArrayList<Attribute> attribute_list,
     String str = String.valueOf(value);
     int length = str.length();
     precision_list.add(length);
-    if (value instanceof Integer) {
-      attribute_list.add(new Attribute(key, "INT", Boolean.FALSE));
-      scale_list.add(0);
-    } else if (value instanceof Double) {
+    /**
+     * An attribute's values could be int for some vertices,
+     * and float for others (e.g., pagerank score).
+     * As a workaround, all integers are treated as float.
+     * Will change it back after supporting retrieving query schema.
+     */
+    if ((value instanceof Double) || (value instanceof Integer)) {
```

Comment:
At line 161, do we need to consider negative double value?

Reply:
there's no difference for negative double values, it's used to record how many digits after "."